tests/benchmark-runner-tests.mjs

```diff
@@ -0,0 +1,323 @@
+            });
+
+            it("should run all test suites", async () => {
+                assert.calledTwice(_runSuiteStub);
+            });
+
+            it("should remove the previous frame and then the current frame", () => {
+                assert.calledTwice(_removeFrameStub);
+            });
+
+            it("should fire the function responsible for finalizing results", () => {
+                assert.calledOnce(_finalizeStub);
+            });
+        });
+
+        describe("_runSuite", () => {
+            let _prepareSuiteStub, _runTestAndRecordResultsStub;
+
+            const suite = SUITES_FIXTURE[0];
+
+            before(async () => {
+                _prepareSuiteStub = stub(runner, "_prepareSuite").callsFake(
+                    () => null
+                );
+
+                _runTestAndRecordResultsStub = stub(
+                    runner,
+                    "_runTestAndRecordResults"
+                ).callsFake(() => null);
+
+                runner._runSuite(suite);
+            });
+
+            it("should prepare the suite first", async () => {
+                assert.calledOnce(_prepareSuiteStub);
+            });
+
+            it("should run and record results for every test in suite", async () => {
+                assert.calledThrice(_runTestAndRecordResultsStub);
+            });
+        });
+    });
+
+    describe("Test", () => {
+        describe("_runTestAndRecordResults", () => {
+            let _runTestSpy;
+
+            const suite = SUITES_FIXTURE[0];
+
+            before(async () => {
+                await runner._appendFrame();
+
+                _runTestSpy = spy(runner, "_runTest");
+
+                await runner._runTestAndRecordResults(suite, suite.tests[0]);
+            });
+
+            it("should run the test with the correct arguments", () => {
+                assert.calledWith(
+                    _runTestSpy,
+                    suite,
+                    suite.tests[0],
+                    runner._page
+                );
+            });
+
+            it("should run client pre and post hooks if present", () => {
+                assert.calledWith(
+                    runner._client.willRunTest,
+                    suite,
+                    suite.tests[0]
+                );
+                assert.calledWith(
+                    runner._client.didRunTest,
+                    suite,
+                    suite.tests[0]
+                );
+            });
+        });
+
+        describe("_runTest", () => {
+            let _writeMarkSpy, _testFnSpy, page;
+            const callback = stub();
+            const suite = SUITES_FIXTURE[0];
+
+            before(async () => {
+                page = { _frame: await runner._appendFrame() };
+                _writeMarkSpy = spy(runner, "_writeMark");
+                _testFnSpy = suite.tests[0].run.callsFake(() => null);
+
+                stubPerformanceNowCalls(8000000, 10000000, 12000000, 13000000);
```

Comment:
Do we really need to use these very large numbers?
How about something like 8000 to make it easier to read?

Reply:
Yeah definitely not necessary, updated :)